src/InfluxDB.LineProtocol/LineProtocolWriter.cs

```diff
@@ -0,0 +1,215 @@
+            textWriter.Write(value.ToString(CultureInfo.InvariantCulture));
+            textWriter.Write('i');
+
+            Position = WriterPosition.FieldWriten;
+
+            return this;
+        }
+
+        public LineProtocolWriter Field(string name, string value)
+        {
+            WriteFieldKey(name);
+            textWriter.Write('=');
+            textWriter.Write('"');
+            textWriter.Write(value.Replace("\"", "\\\""));
+            textWriter.Write('"');
+
+            Position = WriterPosition.FieldWriten;
+
+            return this;
+        }
+
+        public LineProtocolWriter Field(string name, bool value)
+        {
+            WriteFieldKey(name);
+            textWriter.Write('=');
+            textWriter.Write(value ? 't' : 'f');
+
+            Position = WriterPosition.FieldWriten;
+
+            return this;
+        }
+
+        public LineProtocolWriter Timestamp(long value)
+        {
+            switch (Position)
+            {
+                case WriterPosition.FieldWriten:
+                    textWriter.Write(" ");
+                    break;
+                default:
+                    throw new InvalidOperationException();
+            }
+
+            textWriter.Write(value.ToString(CultureInfo.InvariantCulture));
+
+            return this;
+        }
+
+        public LineProtocolWriter Timestamp(TimeSpan value)
+        {
+            return Timestamp(value.Ticks * 100L);
+        }
+
+        public LineProtocolWriter Timestamp(DateTime value)
+        {
+            return Timestamp(value - UnixEpoch);
+        }
+
+        public override string ToString()
+        {
+            if (textWriter is StringWriter stringWriter)
+            {
+                return stringWriter.GetStringBuilder().ToString();
+            }
+
+            return $"{GetType().Name} at postion {this.Position}";
+        }
+
+        private void WriteFieldKey(string name)
+        {
+            switch (Position)
+            {
+                case WriterPosition.MeasurementWriten:
+                case WriterPosition.TagWriten:
+                    textWriter.Write(" ");
+                    break;
+                case WriterPosition.FieldWriten:
+                    textWriter.Write(",");
+                    break;
+                default:
+                    throw new InvalidOperationException();
+            }
+
+            textWriter.Write(EscapeName(name));
+        }
+
+        public static string EscapeName(string nameOrKey)
+        {
+            if (nameOrKey == null) throw new ArgumentNullException(nameof(nameOrKey));
+            return nameOrKey
+                .Replace("=", "\\=")
+                .Replace(" ", "\\ ")
+                .Replace(",", "\\,");
+        }
+
+        public enum WriterPosition
+        {
+            StartOfLine,
+            MeasurementWriten,
+            TagWriten,
+            FieldWriten,
+            TimestampWriten
```

Comment:
Conceptually `StartOfLine` is when we have written `\n`. As we only write `\n` lazily it is only ever used as the starting state when nothing has been written. I will rename it to `NothingWritten`.